src/snowcli/cli/common/snow_cli_typer.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+from functools import wraps
+from typing import Any, Callable, Dict, Optional, Type, Union
+
+from click import ClickException
+from typer import Typer
+from typer.core import TyperCommand, TyperGroup, MarkupMode
+from typer.models import (
+    CommandFunctionType,
+    Default,
+)
+
+from snowcli.cli.common.snow_cli_global_context import snow_cli_global_context_holder
+
+
+def _are_tracebacks_enabled() -> bool:
+    return (
+        snow_cli_global_context_holder.get_global_context_copy().logging_and_exception_handling.enable_tracebacks
+    )
+
+
+def _with_exception_handling(f, *args, **kwargs):
+    try:
+        f(*args, **kwargs)
+    except ClickException as exception:
+        raise exception from None
+    except Exception as exception:
+        if _are_tracebacks_enabled():
+            raise exception from None
+        else:
+            raise ClickException(
+                "An unexpected exception occurred:\n"
+                + f"{exception.__class__.__name__}: {exception}\n"
+                + "Use --debug option to see the full traceback."
+            ) from None
+
+
+class SnowCliTyper(Typer):
```

Comment:
Can you document the class? Why we introduce it? How it differes from parent? Should we add also docs for the methods?

Reply:
Ok, no problem.

Just one question. I believe the questions are only the topic of documentation, they are not your questions to me, right?

Reply:
As we agreed to not use the solution from this PR, I made changes from this comment in new PR: https://github.com/Snowflake-Labs/snowcli/pull/192